core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,190 @@
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.floatValue();
+      case DOUBLE:
+        Preconditions.checkArgument(defaultValue.isNumber(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.doubleValue();
+      case DECIMAL:
+        Preconditions.checkArgument(defaultValue.isNumber(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.decimalValue();
+      case STRING:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.textValue();
+      case UUID:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return UUID.fromString(defaultValue.textValue());
+      case DATE:
+      case TIME:
+      case TIMESTAMP:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return Literal.of(defaultValue.textValue()).to(type).value();
+      case FIXED:
+        Preconditions.checkArgument(
+            defaultValue.isTextual() && defaultValue.textValue().length() == ((Types.FixedType) type).length() * 2,
+            "Cannot parse %s to a %s value",
+            defaultValue,
+            type);
+        byte[] fixedBytes = BaseEncoding.base16().decode(defaultValue.textValue().toUpperCase(Locale.ROOT));
+        return ByteBuffer.allocate(((Types.FixedType) type).length()).put(fixedBytes);
+      case BINARY:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        byte[] binaryBytes = BaseEncoding.base16().decode(defaultValue.textValue().toUpperCase(Locale.ROOT));
+        return ByteBuffer.wrap(binaryBytes);
+      case LIST:
+        Preconditions.checkArgument(defaultValue.isArray(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        List<Object> defaultList = Lists.newArrayList();
+        for (JsonNode element : defaultValue) {
+          defaultList.add(parseDefaultFromJson(type.asListType().elementType(), element));
+        }
+        return defaultList;
+      case MAP:
+        Preconditions.checkArgument(
+            defaultValue.isObject() && defaultValue.has("keys") && defaultValue.has("values") &&
+                defaultValue.get("keys").isArray() && defaultValue.get("values").isArray(),
+            "Cannot parse %s to a %s value",
+            defaultValue,
+            type);
+        Map<Object, Object> defaultMap = Maps.newHashMap();
+        JsonNode keys = defaultValue.get("keys");
+        JsonNode values = defaultValue.get("values");
+        List<JsonNode> keyList = Lists.newArrayList(keys.iterator());
+        List<JsonNode> valueList = Lists.newArrayList(values.iterator());
```

Comment:
It shouldn't be necessary to copy these into lists. Instead, you can iterate over them simultaneously after checking that the size is the same:

```java
  ImmutableMap.Builder<Object, Object> mapBuilder = ImmutableMap.builder();

  Iterator<JsonNode> keyIter = keys.iterator();
  Type keyType = type.asMapType().keyType();
  Iterator<JsonNode> valueIter = values.iterator();
  Type valueType = type.asMapType().valueType();

  while (keyIter.hasNext()) {
    mapBuilder.put(fromJson(keyType, keyIter.next()), fromJson(valueType, valueIter.next()));
  }

  return mapBuilder.build();
```

Reply:
done